subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/TransformAction.java

```diff
@@ -16,20 +16,41 @@
 
 package org.gradle.api.artifacts.transform;
 
+import org.gradle.api.Action;
 import org.gradle.api.Incubating;
 
 import javax.inject.Inject;
 
 /**
  * Interface for artifact transform actions.
  *
- * <p>Implementations must provide a public constructor.</p>
+ * <p>
+ *     A transform action implementation is an abstract class implementing the {@link #transform(TransformOutputs)} method.
+ *     The abstract methods will be created by Gradle when possible.
+ *     A minimal implementation may look like this:
+ * </p>
  *
- * <p>Implementations can receive parameters by using annotated abstract getter methods.</p>
+ * <pre>
+ * public abstract class MyTransform implements TransformAction&lt;TransformParameters.None&gt; {
+ *     {@literal @}InputArtifact
+ *     abstract Provider&lt;FileSystemLocation&gt; getInputArtifact();
  *
- * <p>A property annotated with {@link InputArtifact} will receive the <em>input artifact</em> location, which is the file or directory that the transform should be applied to.
+ *     {@literal @}Override
+ *     public void transform(TransformOutputs outputs) {
+ *         File input = getInputArtifact().get().getAsFile();
+ *         File output = outputs.file(input.getName() + ".transformed");
+ *         // Do something to generate output from input
+ *     }
+ * }
+ * </pre>
  *
- * <p>A property annotated with {@link InputArtifactDependencies} will receive the <em>dependencies</em> of its input artifact.
+ * <ul>
+ *     <li>Do not implement {@link #getParameters()} in your class, the method will be implemented by Gradle.</li>
+ *     <li>Implementations must provide a public constructor.</li>
```

Comment:
What about parameters to the constructor? Shall we elaborate about `@Inject` constructors here, or demand a default constructor instead?

Reply:
I wouldn't mention constructor parameters. How about requiring a default constructor?

Reply:
Let's go with that then.